Detectors/TRD/workflow/include/TRDWorkflow/TRDGlobalTrackingQCSpec.h

```diff
@@ -19,17 +19,113 @@
 // input TRD tracks, TRD tracklets, TRD calibrated tracklets, ITS-TPC tracks, TPC tracks
 // output QC histograms
 
+#include "Headers/DataHeader.h"
 #include "Framework/DataProcessorSpec.h"
+#include "Framework/Task.h"
+#include "Framework/ConfigParamRegistry.h"
+#include "DPLUtils/MakeRootTreeWriterSpec.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "DetectorsBase/Propagator.h"
+#include "DataFormatsParameters/GRPObject.h"
 #include "ReconstructionDataFormats/GlobalTrackID.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "TRDQC/Tracking.h"
+
+#include "DetectorsBase/GRPGeomHelper.h"
 
 using namespace o2::framework;
+using namespace o2::globaltracking;
+using GTrackID = o2::dataformats::GlobalTrackID;
 
 namespace o2
 {
 namespace trd
 {
-/// create a processor spec
-framework::DataProcessorSpec getTRDGlobalTrackingQCSpec(o2::dataformats::GlobalTrackID::mask_t src);
+
+class TRDGlobalTrackingQC : public Task
+{
+ public:
+  TRDGlobalTrackingQC(std::shared_ptr<DataRequest> dr, std::shared_ptr<o2::base::GRPGeomRequest> gr) : mDataRequest(dr), mGGCCDBRequest(gr) {}
+  ~TRDGlobalTrackingQC() override = default;
+  void init(InitContext& ic) final { o2::base::GRPGeomHelper::instance().setRequest(mGGCCDBRequest); }
+  void run(ProcessingContext& pc) final
+  {
+    RecoContainer recoData;
+    recoData.collectData(pc, *mDataRequest.get());
+    updateTimeDependentParams(pc); // Make sure this is called after recoData.collectData, which may load some conditions
+    mQC.reset();
+    mQC.setInput(recoData);
+    mQC.run();
+    pc.outputs().snapshot(Output{"TRD", "TRACKINGQC", 0, Lifetime::Timeframe}, mQC.getTrackQC());
+  }
+  void endOfStream(framework::EndOfStreamContext& ec) final {}
+  void finaliseCCDB(framework::ConcreteDataMatcher& matcher, void* obj) final
+  {
+    if (o2::base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+      return;
+    }
+  }
+
+ private:
+  void updateTimeDependentParams(framework::ProcessingContext& pc)
+  {
+    o2::base::GRPGeomHelper::instance().checkUpdates(pc);
+    static bool initOnceDone = false;
```

Comment:
Does this need to be static or can it be turned into a member var?

Reply:
I think you are right it could also be a member variable. At the point when this DPL CCDB mechanism was added to all workflows it was probably easier to just make it `static` in all devices. If you grep for it you can see that it's done like this everywhere, so I would just leave it like that here as well.